### PR TITLE
refactor(merchant_account_v2): remove routing algorithms from merchant account and add version column

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] Refactoring

## Description

Remove the routing algorithm from merchant_account for api v2. This PR also adds version column to the merchant account.

### Additional Changes

- [x] This PR modifies the database schema
Remove the `routing_algorithm` field from merchant_account. This is for v2 version only.
For v1 version of merchant account, version column has been added.

## How did you test it?

Test the sanity of merchant account

## Impact
Merchant account

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

**Review by SanchithHegde — APPROVED** [review]

File: crates/hyperswitch_domain_models/src/merchant_account.rs [hunk 1]
```diff
@@ -135,7 +135,6 @@ pub struct MerchantAccountSetter {
     pub publishable_key: String,
     pub storage_scheme: MerchantStorageScheme,
     pub metadata: Option<pii::SecretSerdeValue>,
-    pub routing_algorithm: Option<serde_json::Value>,
     pub frm_routing_algorithm: Option<serde_json::Value>,
```
Comment (jarnura) [inline crates/hyperswitch_domain_models/src/merchant_account.rs:138]: remove these this algorithm also

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — DISMISSED** [review]

**Review by SanchithHegde — DISMISSED** [review]

File: crates/hyperswitch_domain_models/src/customer.rs [hunk 2]
```diff
@@ -36,7 +35,6 @@ pub struct Customer {
     pub address_id: Option<String>,
     pub default_payment_method_id: Option<String>,
     pub updated_by: Option<String>,
-    pub version: ApiVersion,
```
Comment (sahkal) [inline crates/hyperswitch_domain_models/src/customer.rs:39]: why are these deleted, please add them back

Comment (Narayanbhat166) [reply crates/hyperswitch_domain_models/src/customer.rs:39]: The idea is to have a single place to set this value, and this should be set at the database layer only. But since we might need this in logs in the future, will add this back

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by Aprabhat19 — APPROVED** [review]
